### PR TITLE
Preserve scaling factors on transforms

### LINKED ISSUE #556: Pass `scale` vector to `mat_decompose` when a `scale_XXX` setter is called on a transform
This will help preserve the `scale` vector in a roundtrip from components -> matrix -> components.

This should resolve the problem observed in #553

This requires a fix in pylinalg, see https://github.com/pygfx/pylinalg/issues/81

### PR DESCRIPTION
**Pull request by Korijn**
Closes #556

### CODE REVIEW

File: pygfx/utils/transform.py [hunk 1]
```diff
@@ -658,3 +661,7 @@ def __matmul__(self, other):
             return RecursiveTransform(other, parent=self)
         else:
             return np.asarray(self) @ other
+
+    @cached
+    def _decomposed(self) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+        return la.mat_decompose(self.matrix)
```
Comment (almarklein) [inline pygfx/utils/transform.py:667]: Scale factors not not needed here?

File: tests/objects/test_world_object.py [hunk 2]
```diff
@@ -390,3 +391,19 @@ def test_bounding_box():
...
+    ob.local.scale = s
+    # without scale preservation in matrix compose -> decompose roundtrip
+    # ob.local.scale becomes (-1, 2, 3)
+    npt.assert_array_equal(ob.local.scale, s)
+
+    child = gfx.WorldObject()
+    ob.add(child)
+    npt.assert_array_equal(child.local.scale, [1, 1, 1])
+    # we don't preserve the WORLD scale component
+    npt.assert_array_equal(child.world.scale, [-1, 2, 3])
```
Comment (almarklein) [inline tests/objects/test_world_object.py:409]: What does this mean in practice? That it's fine to flip the scale factors of e.g. the camera (which is the most common scenario, I think), but not of objects that (may) have children?

File: pygfx/utils/transform.py (hunk 1 above)
Comment (Korijn) [reply pygfx/utils/transform.py:667]: This is on Recursive transform, a subclass of Affine transform. We use this for world transforms.

File: tests/objects/test_world_object.py (hunk 2 above)
Comment (Korijn) [reply tests/objects/test_world_object.py:409]: It means that we can guarantee scaling will be preserved if you set it on local.scale, but not if you set it on world.scale.

Comment (almarklein) [reply tests/objects/test_world_object.py:409]: But this test also seems to show that negative scale factors don't transfer (correctly) to their children, right?

Comment (Korijn) [reply tests/objects/test_world_object.py:409]: Well, they're equivalent, so I wouldn't say they're incorrect.

Comment (almarklein) [reply tests/objects/test_world_object.py:409]: Right, and that's ok, because in this case there is no roundtrip.

**Review by almarklein — APPROVED** [review]

File: tests/objects/test_world_object.py (hunk 2 above)
Comment (Korijn) [reply tests/objects/test_world_object.py:409]: Exactly!

**Comment by Korijn** [comment]
To be clear; I'm sure we _could_ propagate scaling signs down, but I don't think gives any benefits :)

**Pull request merged**